lib/faraday/adapter/net_http_persistent.rb

```diff
@@ -50,7 +50,26 @@ def proxy_uri(env)
       end
 
       def perform_request(http, env)
-        http.request env[:url], create_request(env)
+        if env[:request].stream_response?
+          size = 0
+          yielded = false
+
+          http_response = http.request(env[:url], create_request(env)) do |response|
+            response.read_body do |chunk|
+              if chunk.bytesize.positive? || size.positive?
+                yielded = true
+                size += chunk.bytesize
+                env[:request].on_data.call(chunk, size)
+              end
+            end
+          end
+
+          env[:request].on_data.call(+"", 0) unless yielded
+          http_response.body = nil
+          http_response
+        else
+          http.request(env[:url], create_request(env))
+        end
```

Comment:
This approach is based on https://github.com/lostisland/faraday-net_http/blob/9534fd19bd4898f28361ea60dbc3867edadc15ad/lib/faraday/adapter/net_http.rb#L106-L120

Though I'm happy to hear if you think there is a better way :)